### PR TITLE
Phil

### PR DESCRIPTION
**Pull request by phil8192**
* dockerised OTP
* dockerised propeR
* pre built docker images at: https://cloud.docker.com/u/datasciencecampus/repository/list

### CODE REVIEW

File: README_DOCKER.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+### propeR
+
+Can run propeR from within R session, or alternatively can run our docker
+image.
+
+Put source and destination `.csv` data in a directory, e.g., `/tmp/data/`.
+Example data can be found in `propeR/inst/extdata/`. then:
+
+```
+docker run -v /tmp/data:/mnt datasciencecampus/dsc_proper:1.0 'otp.host="192.168.0.2", fun="pointToPoint", src_file="/mnt/origin.csv", dst_file="/mnt/destination.csv", output.dir="/mnt", startDateAndTime="2019-08-02 12:00:00"'
```
Comment (mshodge) [inline README_DOCKER.md]: You may need to walk me though this as I can't get this step to work. I used a 1.1.1.1 proxy if that's causing issues here. Can get R to connect to my local server, but not using docker

File: README_DOCKER.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+See [Dockerfile](Dockerfile) for more info + deps.
+
+
+A stand-alone OTP server can be built deployed in the [otp/](otp/) directory.
+
+## docker.io
+
+Our images can also be obtained directly from [docker.io](https://docker.io).
+
+* [datasciencecampus/repository/list](https://cloud.docker.com/u/datasciencecampus/repository/list).
```
Comment (mshodge) [inline README_DOCKER.md]: Getting a 404 error for the URL

File: README_DOCKER.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## Running
+
+### OTP
+
+First fire up OTP server (parse `-d` flag to daemonise).
+
+```
+docker run -p 8080:8080 datasciencecampus/dsc_otp:1.0
```
Comment (mshodge) [inline README_DOCKER.md]: Am I right in thinking we can upload any graph, not just the Cardiff one? And if so, can you show me :)

File: README_DOCKER.md (hunk 1 above)
Comment (phil8192) [reply README_DOCKER.md]: hey @mshodge thanks for checking this out.

there is an issue with docker running on windows machines where the container can not access localhost. to accommodate windows users, a work around is to have the container connect to an external interface. when i tested this, `192.168.0.2` was my external interface (at home) - on work macbook/ substitute the `192..` with your external ip which likely starts with `17..`. can find it with:

```
/sbin/ifconfig |grep inet |awk '{print $2}'
```

File: README_DOCKER.md (hunk 2 above)
Comment (phil8192) [reply README_DOCKER.md]: oops. need to be logged in for that url. my bad.

File: README_DOCKER.md (hunk 3 above)
Comment (phil8192) [reply README_DOCKER.md]: at the minute the container is pre-built with the cardiff graph.

it is possible to build a container with a different graph by editing the `build.sh` script in the `otp/` dir. currently set to:

```
GTFS="https://a2s-gtfs.s3.eu-west-2.amazonaws.com/Cardiff/Mar19/Cardiff_Mar19.zip"
OSM="https://github.com/datasciencecampus/access-to-services/raw/master/propeR/data/osm/Cardiff_osm.osm"
```

just issue a `docker build .` in `otp/`.

could parameterise the container, or have it download and build graph every time started. - would be a nicer approach. idea here is to help someone get up and running quickly. i'll add a ticket for this.

Comment (phil8192) [reply README_DOCKER.md]: #28

File: README_DOCKER.md (hunk 1 above)
Comment (mshodge) [reply README_DOCKER.md]: Brilliant. Works now. Will add the additional IP steps to the walkthrough.

File: README_DOCKER.md (hunk 2 above)
Comment (mshodge) [reply README_DOCKER.md]: No worries

File: README_DOCKER.md (hunk 3 above)
Comment (mshodge) [reply README_DOCKER.md]: Thanks, this is great!

**Pull request merged**